app/addons/documents/header/header.react.jsx

```diff
@@ -0,0 +1,245 @@
+          isSelectAllPossible = this.state.isSelectAllPossible,
+          areDocumentsCollapsed = this.state.areDocumentsCollapsed;
+
+      return (
+        <div className='alternative-header'>
+          <ToggleHeaderButton
+            fonticon={'fonticon-select-all'}
+            toggleCallback={this.selectAllDocuments}
+            innerClasses={''}
+            containerClasses={baseClass + 'control-select-all'}
+            text={''}
+            setEnabledClass={!isSelectAllPossible}
+            disabled={!isSelectAllPossible}
+            title={'Select all Documents'} />
+
+          <ToggleHeaderButton
+            fonticon={'fonticon-deselect-all'}
+            toggleCallback={this.deSelectAllDocuments}
+            innerClasses={''}
+            containerClasses={baseClass + 'control-de-select-all'}
+            text={''}
+            setEnabledClass={!isDeselectPossible}
+            disabled={!isDeselectPossible}
+            title={'Deselect all Documents'} />
+
+          <ToggleHeaderButton
+            fonticon={'fonticon-collapse'}
+            toggleCallback={this.toggleCollapseDocuments}
+            innerClasses={''}
+            containerClasses={baseClass + 'control-collapse'}
+            text={''}
+            setEnabledClass={areDocumentsCollapsed}
+            disabled={areDocumentsCollapsed}
+            title={'Collapse all'} />
+
+          <ToggleHeaderButton
+            fonticon={'fonticon-expand'}
+            toggleCallback={this.toggleCollapseDocuments}
+            innerClasses={''}
+            containerClasses={baseClass + 'control-expand'}
+            text={''}
+            setEnabledClass={!areDocumentsCollapsed}
+            disabled={!areDocumentsCollapsed}
+            title={'Expand all'} />
+
+          <ToggleHeaderButton
+            fonticon={'fonticon-trash'}
+            toggleCallback={this.deleteSelected}
+            innerClasses={''}
+            containerClasses={baseClass + 'control-delete'}
+            text={''}
+            title={'Delete selected'} />
+
+          <ToggleHeaderButton
+            fonticon={''}
+            toggleCallback={this.cancelView}
+            innerClasses={''}
+            containerClasses={baseClass + 'control-cancel'}
+            text={'Cancel'}
+            title={'Switch to other view'} />
+        </div>
+      );
+    },
+
+    toggleCollapseDocuments: function () {
+      Actions.toggleCollapseDocuments();
+    },
+
+    selectAllDocuments: function () {
+      Actions.toggleSelectAllDocuments(false);
+    },
+
+    deSelectAllDocuments: function () {
+      Actions.toggleSelectAllDocuments(true);
+    },
+
+    cancelView: function () {
+      Actions.toggleHeaderControls();
+    },
+
+    deleteSelected: function () {
+      Actions.deleteSelected();
+    }
+  });
+
+  var HeaderBarController = React.createClass({
+    getStoreState: function () {
+      return {
+        isToggled: headerBarStore.getStatus(),
```

Comment:
Pretty minor thing. But `getStatus` doesn't mean much. Maybe make it `getToggleStatus`. That explains it more.